saskatoon/member/autocomplete.py

```diff
@@ -30,26 +35,35 @@ def get_queryset(self):
         return qs.distinct()
 
 
+@typechecked
 class ContactAutocomplete(PersonAutocomplete):
     """Persons with contact role (aka Organizations' contact persons)"""
 
-    def __init__(self):
+    def __init__(self) -> None:
         super().__init__(['contact'])
 
 
-class AuthUserAutocomplete(autocomplete.Select2QuerySetView):
+@typechecked
+class AuthUserAutocomplete(Autocomplete):
     """AuthUser autocomplete with optional role filter"""
 
-    def __init__(self, roles=[]):
+    def __init__(self, roles=[]) -> None:
         self.roles = roles
 
     @staticmethod
-    def get_roles_queryset(queryset, roles):
+    def get_roles_queryset(queryset, roles) -> QuerySet[Group]:
         groups = Group.objects.filter(name__in=roles).values('id')
         return queryset.filter(groups__in=groups).distinct()
 
+    """
+    I (Patrick) could not get this method to typecheck properly, mypy does not recognize
+    AuthUser, it calls it AbstractBaseUser and the qs = self.get_roles_queryset line
+    causes since it's already initiated as a QuerySet[AbstractBaseUser], but I'm
+    not sure if the Q invocations after are used on qs in the case of self.q, so I've
+    left it untyped for now.
+    """
```

Comment:
does `-> Queryset[AbstractBaseUser]` work? If not no worries I don't think it's a big deal to leave it untyped

(Also no need to leave your name in comments, we can use `git blame`)